src/exec.js

```diff
@@ -12,6 +12,33 @@ const exec = (command, args, stdin, cwd) => {
 
     const process = childProcess.spawn(command, args, spawnOptions)
 
+    // All of these handlers can close the Promise, so guard closing it twice.
+    let promiseIsClosed = false
```

Comment:
Could we rename this to express better what it means?

`isStdinClosed` or `stdinClosedEarly`?

Reply:
Maybe `promiseIsRejected`? I should've used better wording here; this is about making sure we don't call `reject()` on the Promise twice.